src/ucp/core/ucp_context.c

```diff
@@ -1626,3 +1626,25 @@ ucp_memory_type_detect_mds(ucp_context_h context, void *address, size_t size)
     /* Memory type not detected by any memtype MD - assume it is host memory */
     return UCS_MEMORY_TYPE_HOST;
 }
+
+uint64_t ucp_context_tl_bitmap(ucp_context_h context, const char *dev_name)
+{
+    uint64_t        tl_bitmap;
+    ucp_rsc_index_t tl_idx;
+
+    if (dev_name == NULL) {
+        return context->tl_bitmap;
+    }
+
+    tl_bitmap = 0;
+
+    ucs_for_each_bit(tl_idx, context->tl_bitmap) {
```

Comment:
why not regular for?
context->tl_bitmap is full mask, no gaps

Reply:
this is how we use it everywhere... at least unified mode can add gaps

Reply:
so, looked at tl_bitmap again: it's not a full mask, it's a cached bitmap of resources used by workers, some may be removed due to UNIFIED_MODE=y
so using this map should be OK in general because it will not return resources which are not used by workers anyway